h/api/search.py

```diff
@@ -41,6 +78,11 @@ def build_query(request_params):
         h search API
     :type request_params: webob.multidict.NestedMultiDict
 
+    :param userid: the ID of the authorized user (optional, default: None),
+        if a userid is given then this user's annotations will never be
```

Comment:
This bit seems like too much implementation detail. The argument is the authorized userid, whether/how that factors into a feature like NIPSA seems not related to this interface to me.